Exercise_4.java

```diff
@@ -5,7 +5,44 @@
     // Second subarray is arr[m+1..r] 
     void merge(int arr[], int l, int m, int r) 
     {  
-       //Your code here  
+       //Your code here 
+        int n1 = m - l + 1;

```

Comment:
Good Job! Please add time and space complexity and comments explaining your approach.